fastlane_bot/helpers/txhelpers.py

```diff
@@ -375,13 +376,20 @@ def validate_and_submit_transaction(
         else:
             current_gas_price = arb_tx["gasPrice"]
 
+        signed_arb_tx = self.sign_transaction(arb_tx)
+
+
         # Multiply expected gas by 0.8 to account for actual gas usage vs expected.
         gas_cost_eth = (
             Decimal(str(current_gas_price))
             * Decimal(str(gas_estimate))
             * Decimal(self.ConfigObj.EXPECTED_GAS_MODIFIER)
             / Decimal("10") ** Decimal("18")
         )
+        if self.ConfigObj.network.HAS_LAYER_ONE_GAS_FEE:
+            layer_one_gas_fee = asyncio.get_event_loop().run_until_complete(self.get_layer_one_gas_fee(signed_transaction=signed_arb_tx))
```

Comment:
Looks pretty suspicious (`run_until_complete` in particularly).
Have you tested that?
Have you checked the runtime impact and the implications of this?

Reply:
I have tested that this works.

It's running inside the Main loop if that's what you mean? 

The issue is that if we need to make 3 contract calls to be able to calculate the L1 gas fee - if we don't use async it will be very slow. The function this is in is not asynchronous - so we need to use asyncio to call an async function.

If we put this in a different place, we will make additional network calls when they aren't necessary - and it won't speed up the cycle. 

If you have an alternative that you believe is faster I'm more than happy to discuss.

Reply:
The function is taking around 1-1.18 seconds. If you have any ideas on how to speed it up I'm happy to discuss!

Reply:
> If we put this in a different place, we will make additional network calls when they aren't necessary - and it won't speed up the cycle.

Not if you condition it in the same way it is conditioned here

---

> The function is taking around 1-1.18 seconds. If you have any ideas on how to speed it up I'm happy to discuss!

And do those 1-1.8 seconds take place only once per bot iteration?
Or do they occur within some other lower-level iteration, which means that they will repeat several times on every bot iteration)?

Reply:
It doesn't need to occur once per bot iteration - it's only necessary when we get to this point in the code. It will only occur once per cycle.